Instructions for migrating Kaptain to 2.0.0 (#4352)
mesosphere/dcos-docs-site: pages/dkp/kaptain/2.0.0/migration/index.md

---
layout: layout.pug
navigationTitle: Migration
title: Migrate Kaptain
menuWeight: 7
excerpt: Migrate to Kaptain 2.0.0 on your cluster
beta: false
enterprise: false
---

> Migrate from Kaptain 1.x installation to 2.0.0 on DKP 2.x.

Reviewer: ```suggestion
Migrate from Kaptain 1.3.0 installation to 2.0.0 on DKP 2.x.
```

## Background

> Kaptain 2.0.0 (and later) is installed with Helm, rather than KUDO. As a result, migrating currently requires that you fully remove your prior Kaptain 1.x installation (including deployed models, profiles, notebooks, pipelines, etc.), and install Kaptain 2.0.0. Model artifacts are stored in your object store, therefore, you will not lose these as long as your model store is not deployed to the Kaptain namespace (i.e. you are not using the MinIO server).

Reviewer: ```suggestion
Kaptain 2.0.0 is installed with Helm, rather than KUDO. As a result, migrating currently requires that you fully remove your prior Kaptain 1.x installation (including deployed models, profiles, notebooks, pipelines, etc.), and install Kaptain 2.0.0. Model artifacts are stored in your object store, therefore, you will not lose these as long as your model store is not deployed to the Kaptain namespace (i.e. you are not using the MinIO server).
```

---
layout: layout.pug
navigationTitle: Migration
title: Migrate Kaptain
menuWeight: 7
excerpt: Migrate to Kaptain 2.0.0 on your DKP cluster
beta: false
enterprise: false
---

> Due to a major update in Kaptain's lifecycle management tool, migrating your 1.x installation to 2.x is **not possible**. To ensure that you are up-to-date with the latest features, and can benefit from the most recent performance improvements, we recommend that you uninstall your previous Kaptain version and install the latest available version. 

Reviewer 2: ```suggestion
Due to a major update in Kaptain's lifecycle management tool, migrating your 1.x installation to 2.x is **not possible**. To ensure that you are up-to-date with the latest features, and can benefit from the most recent performance improvements, we recommend that you uninstall your previous Kaptain version and do a fresh install of the latest available version. 
```

> Due to a major update in Kaptain's lifecycle management tool, migrating your 1.x installation to 2.x is **not possible**. To ensure that you are up-to-date with the latest features, and can benefit from the most recent performance improvements, we recommend that you uninstall your Kaptain version and do a fresh install of the latest available version. 

Reviewer 3: ```suggestion
Due to a major update in Kaptain's lifecycle management tooling, migrating your 1.x installation to 2.x is **not possible**. To ensure that you are up-to-date with the latest features, and can benefit from the most recent performance improvements, we recommend that you uninstall your Kaptain version and do a fresh install of the latest available version. 
```

> <p class="message--warning"><strong>WARNING: </strong>For this release, you will have to uninstall and install Kaptain to get to the latest version. When uninstalling Kaptain, all data, the current state, and configuration of your application will be lost.</p>

Reviewer 2: Alternative ```suggestion
<p class="message--warning"><strong>WARNING: </strong>For this release, you will have to uninstall and install Kaptain to get to the latest version. Because this is a fresh install, ensure you back up all of your data before proceeding.</p>
```

## Background

> Previous versions of Kaptain were installed with KUDO. From 2.0.0 and later, you can install Kaptain with Helm. As a result, migrating currently requires that you fully remove your prior Kaptain 1.x installation (including deployed models, profiles, notebooks, pipelines, etc.), and install Kaptain 2.0.0. Model artifacts are stored in your object store, therefore, you will not lose these as long as your model store is not deployed to the Kaptain namespace (that is, you are not using the MinIO server). However, **create a backup** of all your required data.

Reviewer 3: We need instructions on how to both create AND restore a backup....

> Previous versions of Kaptain were installed with KUDO. From 2.0.0 and later, you can install Kaptain with Helm. As a result, migrating currently requires that you fully remove your prior Kaptain 1.x installation (including deployed models, profiles, notebooks, pipelines, etc.), and install Kaptain 2.0.0. Model artifacts are stored in your object store, therefore, you will not lose these as long as your model store is not deployed to the Kaptain namespace (that is, you are not using the MinIO server). However, **create a backup** of all your required data.

Reviewer 2: @chessahoekstra @alembiewski @cneth 
I think this pplies here too: there is a way to back up and restore data. However, product and the team have decided not to document this as there is no way to guarantee it will work (at least for this release). Should we explicitly call this out? or leave a general statement as above? WDYT?

> ## Uninstall Kaptain 1.x

Reviewer 2: ```suggestion
## Uninstall Kaptain
```

> Previous versions of Kaptain were installed with KUDO. From 2.0.0 and later, you can install Kaptain with Helm. As a result, migrating currently requires that you fully remove your prior Kaptain 1.x installation (including deployed models, profiles, notebooks, pipelines, etc.), and install Kaptain 2.0.0. Model artifacts are stored in your object store, therefore, you will not lose these as long as your model store is not deployed to the Kaptain namespace (that is, you are not using the MinIO server). However, **create a backup** of anything you want to keep or migrate to your new Kaptain installation.

Reviewer 2: @timcharper since we are uninstalling and installing from scratch, customer will lose the current state, data and config of Kaptain (AFAIK). But here we say 
"[...], you will not lose these as long as your model store is not deployed to the Kaptain namespace (that is, you are not using the MinIO server). However, **create a backup** of anything you want to keep or migrate to your new Kaptain installation." 
Should we delete the last sentence since backing up is not possible? And Model artifacts CAN be preserved? Just double-checking.

Refer to [Uninstall documentation][uninstall_kap] for instructions on how to uninstall your KUDO-based Kaptain version.

> ## Fresh install of Kaptain 2.x

Reviewer 2: ```suggestion
## Fresh install of Kaptain
```

> ## Install Kaptain 2.x

Reviewer 2: ```suggestion
## Fresh install of Kaptain 2.x
```

---
layout: layout.pug
navigationTitle: Migration
title: Migrate Kaptain
menuWeight: 7
excerpt: Migrate to Kaptain 2.0.0 on your DKP cluster
beta: false
enterprise: false
---

> Migrate your Kaptain 1.x installation to the latest version on DKP 2.x.

Reviewer 2: ```suggestion
Due to a major update in Kaptain's lifecycle management tool, migrating your 1.x installation to 2.x is not possible. To ensure that you are up-to-date with the latest features, and can benefit from the most recent performance improvements, we recommend that you uninstall your previous Kaptain version and install the latest available version. 
```

<p class="message--warning"><strong>WARNING: </strong>For this release, you will have to uninstall and install Kaptain to get to 2.0.0. When uninstalling Kaptain, all data, the current state, and configuration of your application will be lost.</p>

## Background

Previous versions of Kaptain were installed with KUDO. From 2.0.0 and later you can install Kaptain with Helm. As a result, migrating currently requires that you fully remove your prior Kaptain 1.x installation (including deployed models, profiles, notebooks, pipelines, etc.), and install Kaptain 2.0.0. Model artifacts are stored in your object store, therefore, you will not lose these as long as your model store is not deployed to the Kaptain namespace (that is, you are not using the MinIO server). However, **create a backup** of anything you want to keep or migrate to your new Kaptain installation.

## Uninstall Kaptain 1.x

> Refer to [Uninstall documentation] for instructions on how to uninstall your KUDO-based Kaptain installation.

Reviewer 2: ```suggestion
Refer to [Uninstall documentation][uninstall_kap] for instructions on how to uninstall your KUDO-based Kaptain installation.
```

<!-- TODO: Links to 1.3.0/uninstall. -->

## Install Kaptain 2.x

> Refer to [Install Kaptain documentation] for instructions on how to install your Helm-based Kaptain installation.

Reviewer 2: ```suggestion
Refer to [Install Kaptain documentation][install_kap] for instructions on how to install your Helm-based Kaptain installation.
```

> [install_kap]: ../install

Reviewer 2: ```suggestion
[install_kap]: ../install
[uninstall_kap]: ../../1.3.0/uninstall
```
